### PR TITLE
Use topology labels instead of old beta names

### PR DESCRIPTION
**Pull request by thockin**
As per KEP 1659, topology keys are standardize, so we can fix up places to prefer them.  This is part of a large sweep of the codebase.

First this renames some consts to make more sense.  Second it (mechanically) converts all call-sites that *only* used the old labels to use the new.  Any call-site that explicitly used both is left as-is.

/kind cleanup

```release-note
Reminder: The labels "failure-domain.beta.kubernetes.io/zone" and "failure-domain.beta.kubernetes.io/region" are deprecated in favor of "topology.kubernetes.io/zone" and "topology.kubernetes.io/region" respectively.  All users of the "failure-domain.beta..." labels should switch to the "topology..." equivalents.
```

### CODE REVIEW

File: staging/src/k8s.io/api/core/v1/well_known_labels.go [hunk 1]
```diff
@@ -19,10 +19,10 @@ package v1
 const (
 	LabelHostname = "kubernetes.io/hostname"
 
-	LabelZoneFailureDomain       = "failure-domain.beta.kubernetes.io/zone"
-	LabelZoneRegion              = "failure-domain.beta.kubernetes.io/region"
-	LabelZoneFailureDomainStable = "topology.kubernetes.io/zone"
-	LabelZoneRegionStable        = "topology.kubernetes.io/region"
+	LabelFailureDomainBetaZone   = "failure-domain.beta.kubernetes.io/zone"   // deprecated, remove after 1.20
```
Comment (andrewsykim) [inline staging/src/k8s.io/api/core/v1/well_known_labels.go:22]: Is this breaking Go compatibility? I think that's why we kept the beta label pointing to `LabelZoneFailureDomain` and named the new topology labels `LabelZoneFailureDomainStable`.

File: pkg/util/node/node.go [hunk 2]
```diff
@@ -181,22 +181,21 @@ func GetNodeIP(client clientset.Interface, name string) net.IP {
...
 // GetZoneKey will first check failure-domain.beta.kubernetes.io/zone and if not exists, will then check
 // topology.kubernetes.io/zone
+// TODO(thockin): stop supporting "failure-domain.beta..." names after 1.20
 func GetZoneKey(node *v1.Node) string {
 	labels := node.Labels
 	if labels == nil {
 		return ""
 	}
 
-	// TODO: prefer stable labels for zone in v1.18
```
Comment (andrewsykim) [inline pkg/util/node/node.go:190]: Related to this TODO: https://github.com/kubernetes/kubernetes/pull/88493

**Review by andrewsykim — COMMENTED** [review]
Not sure if you wanted an early review, but left some comments.

File: staging/src/k8s.io/api/core/v1/well_known_labels.go (hunk 1 above)
Comment (thockin) [reply staging/src/k8s.io/api/core/v1/well_known_labels.go:22]: It's not clear to me if Go-level compat is a goal or not?

Comment (thockin) [reply staging/src/k8s.io/api/core/v1/well_known_labels.go:22]: Will retain old names

File: pkg/util/node/node.go (hunk 2 above)
Comment (thockin) [reply pkg/util/node/node.go:190]: Will add a note.  After 20 goes, we can just nix it all

**Comment by thockin** [comment]
@msau42 Give https://github.com/kubernetes/kubernetes/issues/92237#issuecomment-719709773 I am very unsure about the last commit here.  It's not clear to me that *any* of these changes in pkg/volume/... or plugin/pkg/admission/storage/persistentvolume/... or staging/src/k8s.io/cloud-provider/volume/... or staging/src/k8s.io/csi-translation-lib/... or test/e2e/storage/... are correct, as well as some changes in staging/src/k8s.io/legacy-cloud-providers/...

Even if this passes tests.  Can you tell me what the plan is?  After 1.20 the older "beta" names are scheduled to be removed, and storage seems to be a large fraction of it.

**Comment by msau42** [comment]
> After 1.20 the older "beta" names are scheduled to be removed, and storage seems to be a large fraction of it.

In #81431, the plan was to continue using beta labels for volumes, let csi migration handle the translation to csi topology labels, and then remove the beta labels from nodes once csi migration is GA.

The reason for this is because users can keep their PVs throughout upgrade, and they will have beta labels/NodeAffinity on them. If we start creating nodes without beta labels, then those PVs will be broken on a node upgrade without CSI migration.

Unfortnately, CSI migration is slipping, and the last cloud plugin is probably not going to GA until at least 1.24, and this is going to block removal of beta labels on nodes. PV.NodeAffinity is immutable so we can't go in and backfill GA labels on old objects. 

Do you see ways we can decouple csi migration from this? One possibility is to do an in-memory translation of beta to GA labels in the scheduler. It's basically a more generic method of translation instead of the per-plugin csi translation model.

cc @mattcary who is looking at csi migration from the gce side.

**Comment by andrewsykim** [comment]
This is the relevant thread where we agreed to keep the old beta labels around as long as existing PVs had them: https://github.com/kubernetes/kubernetes/pull/81431#discussion_r341216734 

The consensus was that supporting the beta labels longer than we ideally want is simpler than reasoning about backfilling the GA labels on existing PVs. In v1.20 though, we should update the "primary" label to be the GA topology labels, currently they are still pointing to the beta labels:

https://github.com/kubernetes/kubernetes/blob/77866160efce1ce4c3885d436b1cebaedc80c89c/staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go#L57-L80

and

https://github.com/kubernetes/kubernetes/blob/05a46dbb6026dfd9ae11600f8dc08a99994acdec/pkg/util/node/node.go#L190-L200

**Comment by thockin** [comment]
In truth, I am fine leaving the old names around as long as we need, as long as it is reduced to JUST the places that *really* need it and we comment clearly why.  If I had my druthers, we would stop using beta forms for anything new and document that we keep the old labels around JUST for compat with old instances.

Maybe one day we can actually delete the names, but maybe not.

**Comment by msau42** [comment]
Tim and I discussed a bit. We agreed it would be good to get new provisioned volumes using the GA label, so we can reduce the code surface that still uses this. It's a little complicated though because CSI migration will need to be updated to be able to handle both beta and GA labels, so it would be better to handle volumes in a separate pr.

**Comment by thockin** [comment]
I'll revisit these changes and be more surgical.  In my ideal world, the places that use the old name are very few and VERY clearly documented.

**Comment by thockin** [comment]
rebased

This PR now only includes what I think are "obvious" and "safe" changes.  No storage changes.

PTAL

File: pkg/kubelet/apis/well_known_labels.go [hunk 3]
```diff
@@ -34,10 +34,10 @@ const (
 
 var kubeletLabels = sets.NewString(
 	v1.LabelHostname,
-	v1.LabelZoneFailureDomainStable,
-	v1.LabelZoneRegionStable,
-	v1.LabelZoneFailureDomain,
-	v1.LabelZoneRegion,
+	v1.LabelTopologyZone,
```
Comment (andrewsykim) [inline pkg/kubelet/apis/well_known_labels.go:37]: Were we going to keep `LabelZoneFailureDomain` for compatbility?

Comment (andrewsykim) [reply pkg/kubelet/apis/well_known_labels.go:37]: Oh nvm I see the alias now -- wishing I thought of that in the original PR

File: pkg/util/node/node.go [hunk 4]
```diff
@@ -187,16 +187,17 @@ func GetZoneKey(node *v1.Node) string {
...
-	zone, ok := labels[v1.LabelZoneFailureDomain]
+	// TODO: "failure-domain.beta..." names are deprecated, but will
+	// stick around a long time due to existing on old extant objects like PVs.
+	// Maybe one day we can stop considering them (see #88493).
+	zone, ok := labels[v1.LabelFailureDomainBetaZone]
```
Comment (andrewsykim) [inline pkg/util/node/node.go:193]: Would it be safe at this point to at least switch the priority order here so that the topology labels are checked first?

Comment (andrewsykim) [reply pkg/util/node/node.go:193]: Same for regions below

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure.go [hunk 5]
```diff
@@ -727,8 +727,8 @@ func (az *Cloud) SetInformers(informerFactory informers.SharedInformerFactory) {
 		UpdateFunc: func(prev, obj interface{}) {
 			prevNode := prev.(*v1.Node)
 			newNode := obj.(*v1.Node)
-			if newNode.Labels[v1.LabelZoneFailureDomain] ==
-				prevNode.Labels[v1.LabelZoneFailureDomain] {
+			if newNode.Labels[v1.LabelTopologyZone] ==
```
Comment (andrewsykim) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure.go:730]: This switched from beta to GA labels, it's probably safe but I think we probably want to switch this in a separate PR.

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure.go [hunk 6]
```diff
@@ -762,7 +762,7 @@ func (az *Cloud) updateNodeCaches(prevNode, newNode *v1.Node) {
 
 	if prevNode != nil {
 		// Remove from nodeZones cache.
-		prevZone, ok := prevNode.ObjectMeta.Labels[v1.LabelZoneFailureDomain]
+		prevZone, ok := prevNode.ObjectMeta.Labels[v1.LabelTopologyZone]
```
Comment (andrewsykim) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure.go:765]: same here, switched from beta to GA

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure.go [hunk 7]
```diff
@@ -785,7 +785,7 @@ func (az *Cloud) updateNodeCaches(prevNode, newNode *v1.Node) {
 
 	if newNode != nil {
 		// Add to nodeZones cache.
-		newZone, ok := newNode.ObjectMeta.Labels[v1.LabelZoneFailureDomain]
+		newZone, ok := newNode.ObjectMeta.Labels[v1.LabelTopologyZone]
```
Comment (andrewsykim) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure.go:788]: same here, switched from beta to GA

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce.go [hunk 8]
```diff
@@ -737,8 +737,8 @@ func (g *Cloud) SetInformers(informerFactory informers.SharedInformerFactory) {
 		UpdateFunc: func(prev, obj interface{}) {
 			prevNode := prev.(*v1.Node)
 			newNode := obj.(*v1.Node)
-			if newNode.Labels[v1.LabelZoneFailureDomain] ==
-				prevNode.Labels[v1.LabelZoneFailureDomain] {
+			if newNode.Labels[v1.LabelTopologyZone] ==
```
Comment (andrewsykim) [inline staging/src/k8s.io/legacy-cloud-providers/gce/gce.go:740]: This changed from beta->GA which is probably safe but might be good to keep this in a separate PR.

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go [hunk 9]
```diff
@@ -64,7 +64,7 @@ func splitNodesByZone(nodes []*v1.Node) map[string][]*v1.Node {
 }
 
 func getZone(n *v1.Node) string {
-	zone, ok := n.Labels[v1.LabelZoneFailureDomain]
+	zone, ok := n.Labels[v1.LabelTopologyZone]
```
Comment (andrewsykim) [inline staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go:67]: same here, went from beta->GA

File: staging/src/k8s.io/legacy-cloud-providers/vsphere/nodemanager.go [hunk 10]
```diff
@@ -203,8 +203,8 @@ func (nm *NodeManager) DiscoverNode(node *v1.Node) error {
 						node.Name, vm, res.vc, res.datacenter.Name())
 
 					// Get the node zone information
-					nodeFd := node.ObjectMeta.Labels[v1.LabelZoneFailureDomain]
-					nodeRegion := node.ObjectMeta.Labels[v1.LabelZoneRegion]
+					nodeFd := node.ObjectMeta.Labels[v1.LabelTopologyZone]
```
Comment (andrewsykim) [inline staging/src/k8s.io/legacy-cloud-providers/vsphere/nodemanager.go:206]: Switched beta->GA

File: staging/src/k8s.io/legacy-cloud-providers/vsphere/nodemanager.go [hunk 11]
```diff
@@ -203,8 +203,8 @@ func (nm *NodeManager) DiscoverNode(node *v1.Node) error {
...
+					nodeRegion := node.ObjectMeta.Labels[v1.LabelTopologyRegion]
```
Comment (andrewsykim) [inline staging/src/k8s.io/legacy-cloud-providers/vsphere/nodemanager.go:207]: switched beta->GA

File: test/e2e/framework/node/resource.go [hunk 12]
```diff
@@ -518,11 +518,11 @@ func GetClusterZones(c clientset.Interface) (sets.String, error) {
 	// collect values of zone label from all nodes
 	zones := sets.NewString()
 	for _, node := range nodes.Items {
-		if zone, found := node.Labels[v1.LabelZoneFailureDomain]; found {
+		if zone, found := node.Labels[v1.LabelTopologyZone]; found {
```
Comment (andrewsykim) [inline test/e2e/framework/node/resource.go:521]: This should be beta label since line 525 also checks for LabelTopologyZone

File: test/e2e/framework/providers/gce/util.go [hunk 13]
```diff
@@ -36,12 +36,12 @@ func RecreateNodes(c clientset.Interface, nodes []v1.Node) error {
 	for i := range nodes {
 		node := &nodes[i]
 
-		if zone, ok := node.Labels[v1.LabelZoneFailureDomain]; ok {
+		if zone, ok := node.Labels[v1.LabelTopologyZone]; ok {
```
Comment (andrewsykim) [inline test/e2e/framework/providers/gce/util.go:39]: This should be beta label since line 44 also checks for LabelTopologyZone

File: pkg/util/node/node.go (hunk 4 above)
Comment (andrewsykim) [reply pkg/util/node/node.go:193]: On second thought, it might be cleaner to keep this PR scoped to the new constants and do the priority switch in a follow-up PR

**Review by andrewsykim — COMMENTED** [review]
There were a few places where we switch the priority order so that GA labels are checked first and some places where the beta labels are still checked first. I'm thinking it would be cleaner to keep this PR scoped to the new constant names and have a follow-up PR to switch the order we check labels (GA with a fallback to beta).

File: pkg/util/node/node.go (hunk 4 above)
Comment (thockin) [reply pkg/util/node/node.go:193]: If at all?  I am not sure it makes any real sense to switch the priority.  If the labels are the same, no-op.  If they are different, best case is that we break users.

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure.go (hunk 5 above)
Comment (thockin) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure.go:730]: If we want to do that I can just pop the last commit.  That's fair, why don't I do that.

File: test/e2e/framework/node/resource.go (hunk 12 above)
Comment (thockin) [reply test/e2e/framework/node/resource.go:521]: I'll just remove the now-redundant block

File: test/e2e/framework/providers/gce/util.go (hunk 13 above)
Comment (thockin) [reply test/e2e/framework/providers/gce/util.go:39]: will remove

**Comment by thockin** [comment]
Re-pushed with just the mechanical changes.  The actual changes can be a new PR

**Review by andrewsykim — COMMENTED** [review]
/lgtm
